test/helpers/node.go

```diff
@@ -67,12 +70,36 @@ func GetVagrantSSHMeta(vmName string) *SSHMeta {
 		log.Errorf("Node %s not found in ssh config", vmName)
 		return nil
 	}
-
-	return &SSHMeta{
+	sshMeta := &SSHMeta{
 		sshClient: node.GetSSHClient(),
 		rawConfig: config,
 		nodeName:  vmName,
 	}
+
+	sshMeta.setBasePath()
+	return sshMeta
+}
+
+// setBasePath if the SSHConfig is defined we set the BasePath to the GOPATH,
+// from golang 1.9 GOPATH is by default $HOME/go so we also check that.
+func (s *SSHMeta) setBasePath() {
+	if config.CiliumTestConfig.SSHConfig == "" {
+		return
+	}
+
+	gopath := s.Exec("echo $GOPATH").SingleOut()
+	if gopath != "" {
+		BasePath = filepath.Join(gopath, CiliumPath)
+		return
+	}
+
+	home := s.Exec("echo $HOME").SingleOut()
+	if home == "" {
+		return
+	}
+
+	BasePath = filepath.Join(home, "go", CiliumPath)
```

Comment:
Does `BasePath` needs to be public?

Reply:
Yep, it's used by the PolicyGen in Nightly.

Reply:
Seems like `policygen.TestSpec` and `policygen.Target` should be initialised with proper BasePath , and refer to it in their code. Also seems like other places using BasePath are already using SSHMeta, maybe it should also be stored in it? I think we need an issue to track this if this is too big to work on in this PR.

Reply:
I think the above BasePath cleanup should be done in a separate PR this PR is useful even without the cleanup.